firestore/client/CMakeLists.txt

```diff
@@ -0,0 +1,125 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generate the version information from the CMake values.
+configure_file(version_info.h.in version_info.h)
+
+# the client library
+add_library(firestore_client
+        field_path.h
+	field_path.cc
+	)
+target_link_libraries(firestore_client PUBLIC
+        firestore_protos)
+#        gRPC::grpc++ gRPC::grpc protobuf::libprotobuf
+#        PRIVATE firestore_common_options)
+target_include_directories(firestore_client PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
+    $<INSTALL_INTERFACE:include>)
+target_compile_options(firestore_client PUBLIC ${GOOGLE_CLOUD_CPP_EXCEPTIONS_FLAG})
+add_library(firestore::client ALIAS firestore_client)
+
+option(FIRESTORE_CLIENT_CLANG_TIDY
+    "If set compiles the Cloud Firestore client with clang-tidy."
+    "")
+if (CLANG_TIDY_EXE AND FIRESTORE_CLIENT_CLANG_TIDY)
+    message(STATUS "clang-tidy build enabled.")
+    set_target_properties(
+        firestore_client PROPERTIES
+        CXX_CLANG_TIDY "${CLANG_TIDY_EXE}"
+    )
+endif ()
+
+add_library(firestore_client_testing
+        field_path.h
+	field_path_test.cc)
+
+#         testing/chrono_literals.h
```

Comment:
I know you are developing right now and there is a lot of experimentation needed, but please remember to remove the commented out blocks.